src/services/earn/earn-service.ts

```diff
@@ -930,6 +1021,50 @@ export class EarnService implements IEarnService {
     return { quote: bestQuote.quote, tx: bestQuote.tx };
   }
 
+  private buildMigrateTransaction(params: {
+    migrate: MigrateEarnPositionParams;
+    migrationAsset: ViemAddress;
+    caller: Address;
+    vault: ViemAddress;
+    fromPositionId: PositionId;
+  }) {
+    const { migrate, migrationAsset, caller, vault, fromPositionId } = params;
+    if (migrate?.type == MigrateEarnType.CREATE) {
+      const strategyIdBigInt = BigInt(migrate.strategyId.split('-')[2]);
+      const misc: Record<number, Hex> = {
+        [MigrationCode.MIGRATE_FROM_POSITION_AND_CREATE]: encodeAbiParameters(parseAbiParameters('string'), [fromPositionId]),
```

Comment:
Great idea with using the full position id. I had imagined we would have use the numeric position id, but I like your idea a lot better